project/UI/controller.js

```diff
@@ -288,20 +338,27 @@
                 });
             }
             var tempPosts = [];
-            if (checkHashtags(filter)){
-                for (var k=0;k<filter.hashtags.length;k++)
-                {
+            if (typeof filter.hashtag === "string"){
                     for (var i=0;i<newPosts.length;i++) {
                         for (var j = 0; j < newPosts[i].hashtags.length; j++) {
-                            if (newPosts[i].hashtags[j]===filter.hashtags[k])
+                            if (newPosts[i].hashtags[j]===filter.hashtag && newPosts[i].active)
                                 tempPosts.push(newPosts[i]);
                         }
                     }
-                }
                 newPosts = tempPosts;
             }
             newPosts.sort(comporator);
-            return newPosts.slice(skip,top);
+            for (var i = skip;i<skip+top;i++) {
```

Comment:
Firstly, you copypaste the code above, it's a bad practive
Secondly, is it possible to filter active posts and then do ``posts.slice(skip,top) ``?
I mean something like this ``posts.filter(post => post.active).slice(skip, top) ``